mobile/src/navigation/HomeStackNavigator.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { StyleSheet, Image } from 'react-native';
+import { createStackNavigator } from '@react-navigation/stack';
+import { RouteProp } from '@react-navigation/native';
+import { HomeScreen } from '../screens/HomeScreen';
+import { BookScreen } from '../screens/BookScreen';
+import { Book } from '../models/Book';
+
+type HomeStackParams = {
+  Home: undefined;
+  Book: { book: Book };
+};
+
+export type BookScreenRouteProps = RouteProp<HomeStackParams, 'Book'>;
```

Comment:
Damn the TS definitions are annoying for React Navigation, but I think I found a good way to type the BookScreen props. You should be able to get rid of the BookScreenRouteProps type here, and export HomeStackParams, and then...